src/menu_parser.py

```diff
@@ -157,92 +246,44 @@ def __get_self_service_prices(
 
-            if dish[4] == "0":  # Non-Vegetarian
-                # Add a base price to the dish
-                if "Fi" in dish[2]:  # Fish
+            if dish[0] != "Studitopf" and dish[4] == "0":  # Non-Vegetarian
+                if "Fi" in dish[2]:
                     base_price_type = StudentenwerkMenuParser.SelfServiceBasePriceType.FISH
-                # Sausage. TODO: Find better way to distinguish between sausage and meat
+                # TODO: Find better way to distinguish between sausage and meat
                 elif "wurst" in dish_name.lower() or "würstchen" in dish_name.lower():
                     base_price_type = StudentenwerkMenuParser.SelfServiceBasePriceType.SAUSAGE
-                else:  # Meat
+                else:
                     base_price_type = StudentenwerkMenuParser.SelfServiceBasePriceType.MEAT
             else:
-                base_price_type = StudentenwerkMenuParser.SelfServiceBasePriceType.VEGETARIAN
+                base_price_type = StudentenwerkMenuParser.SelfServiceBasePriceType.VEGETARIAN_SOUP_STEW
             return StudentenwerkMenuParser.__get_self_service_prices(base_price_type, price_per_unit_type)
 
-    # Some of the locations do not use the general Studentenwerk system and do not have a location id.
-    # It differs how they publish their menus — probably everyone needs an own parser.
-    # For documentation they are in the list but commented out.
-    location_id_mapping: Dict[str, int] = {
-        "mensa-arcisstr": 421,
-        "mensa-arcisstrasse": 421,  # backwards compatibility
-        "mensa-garching": 422,
-        "mensa-leopoldstr": 411,
-        "mensa-lothstr": 431,
-        "mensa-martinsried": 412,
-        "mensa-pasing": 432,
-        "mensa-weihenstephan": 423,
-        "stubistro-arcisstr": 450,
-        # "stubistro-benediktbeuern": ,
-        "stubistro-goethestr": 418,
-        "stubistro-großhadern": 414,
-        "stubistro-rosenheim": 441,
-        "stubistro-schellingstr": 416,
-        # "stubistro-schillerstr": ,
-        "stucafe-adalbertstr": 512,
-        "stucafe-akademie-weihenstephan": 526,
-        # "stucafe-audimax" ,
-        "stucafe-boltzmannstr": 527,
-        "stucafe-garching": 524,
-        # "stucafe-heßstr": ,
-        "stucafe-karlstr": 532,
-        # "stucafe-leopoldstr": ,
-        # "stucafe-olympiapark": ,
-        "stucafe-pasing": 534,
-        # "stucafe-weihenstephan": ,
-    }
-
-    base_url: str = "http://www.studentenwerk-muenchen.de/mensa/speiseplan/speiseplan_{location_id}_-de.html"
+    base_url: str = "http://www.studentenwerk-muenchen.de/mensa/speiseplan/speiseplan_{canteen_id}_-de.html"
     base_url_with_date: str = (
-        "http://www.studentenwerk-muenchen.de/mensa/speiseplan/speiseplan_{date}_{location_id}_-de.html"
+        "http://www.studentenwerk-muenchen.de/mensa/speiseplan/speiseplan_{date}_{canteen_id}_-de.html"
```

Comment:
Does this also work with the canteen_id (which should be e.g. `mensa-garching`)?
As far as I tested it, it doesn't, but the `url_id` or here previously `location_id` is needed?

Reply:
oh, sorry. The url_id is needed

Reply:
I'll change it